src/net.h

```diff
@@ -36,12 +36,10 @@
 #define MAXBUFSIZE ((MAXSIZE > 2048 ? MAXSIZE : 2048) + 128)
 
 typedef struct vpn_packet_t {
-	struct {
-		unsigned int probe: 1;
-		unsigned int tcp: 1;
-	};
 	uint16_t len;           /* the actual number of bytes in the `data' field */
 	uint8_t data[MAXSIZE];
+	unsigned int probe: 1;
```

Comment:
These bitfields are better at the start of the struct:
- It's more cache friendly to have all parameters close together
- It allows vpn_packet_t to be allocated with a variable size in the future.

Reply:
I actually tried those combinations before and decided to place them at bottom, when placed them top it's 1604 bytes and when placed bottom it's 1600 bytes. If ordering them at top gives us benefits then I'll change it.

Reply:
Ah, you can have the best of both worlds by writing:
```
typedef struct vpn_packet_t {
    uint16_t probe: 1;
    uint16_t tcp: 1;
    uint16_t len;
    uint8_t data[MAXSIZE];
} vpn_packet_t;
```